core/templates/dev/head/pages/exploration_player/AudioControlsDirective.js

```diff
@@ -64,12 +65,19 @@ oppia.directive('audioControls', [
           };
 
           $scope.playPauseAudioTranslation = function() {
-            // TODO(tjiang11): Change from on-demand loading to pre-loading.
-
-            // TODO(tjiang11): On first play, ask learner to pick language
-            // and subsequently for confirmation to use bandwidth 
-            // to download audio files.
+            if (!AudioPreloaderService.hasPreloaded()) {
+              if (getCurrentAudioTranslation()) {
```

Comment:
It should evaluate to true if there is an audio translation available in the current language. 

Yeah, I assumed learners would either be willing to use the necessary bandwidth for all audio or decline to use any bandwidth. But I suppose there is a middleground where some learners may need help with only one specific piece of content and also be restrictive in their use of bandwidth.

Yeah sorry, it was never in the design doc. How about this: clicking "continue" will only download the current file. But a checkbox will also be present saying "Download all audio for this exploration (XX MB)". And preloading will occur only if the learner checks that. Does that sound okay?